lib/rules/no-duplicate-imports.js

```diff
@@ -8,39 +8,188 @@
+ */
+function getImportExportType(node) {
+    if (node && node.specifiers && node.specifiers.length > 0) {
+        const nodeSpecifiers = node.specifiers;
+        const index = nodeSpecifiers.findIndex(
+            ({ type }) =>
+                isImportExportSpecifier(type, "specifier") ||
+                  isImportExportSpecifier(type, "namespace")
+        );
+        const i = index > -1 ? index : 0;
+
+        return nodeSpecifiers[i].type;
+    }
+    if (node.type === "ExportAllDeclaration") {
+        if (node.exported) {
+            return "ExportNamespaceSpecifier";
+        }
+        return "ExportAll";
+    }
+    if (node.type === "ImportDeclaration") {
+        return "SideEffectImport";
+    }
+    return node.type;
+}
+
+/**
+ * Returns a boolean indicates if two (import|export) can be merged
+ * @param {ASTNode} node1 A node to get.
+ * @param {ASTNode} node2 A node to get.
+ * @returns {boolean} True if two (import|export) can be merged, false if they can't.
+ */
+function isImportExportCanBeMerged(node1, node2) {
+    const importExportType1 = getImportExportType(node1);
+    const importExportType2 = getImportExportType(node2);
+
+    if (
+        (importExportType1 === "ExportAll" &&
+              importExportType2 !== "ExportAll" &&
+              importExportType2 !== "SideEffectImport") ||
+          (importExportType1 !== "ExportAll" &&
+              importExportType1 !== "SideEffectImport" &&
+              importExportType2 === "ExportAll")
+    ) {
+        return false;
+    }
+    if (
+        (isImportExportSpecifier(importExportType1, "namespace") &&
+              isImportExportSpecifier(importExportType2, "specifier")) ||
+          (isImportExportSpecifier(importExportType2, "namespace") &&
+              isImportExportSpecifier(importExportType1, "specifier"))
+    ) {
+        return false;
+    }
+    return true;
+}
+
+/**
+ * Returns a boolean if we should report (import/export).
+ * @param {ASTNode} node A node to be reported or not.
+ * @param {[ASTNode]} previousNodes An array contains previous nodes of the module imported or exported.
+ * @returns {boolean} True if the (import/export) should be reported.
+ */
+function shouldReportImportExport(node, previousNodes) {
+    if (previousNodes) {
+        let i = 0;
+
+        while (i < previousNodes.length) {
+            if (isImportExportCanBeMerged(node, previousNodes[i])) {
+                return true;
+            }
+            i++;
+        }
+    }
+    return false;
+}
+
+/**
+ * Returns array contains only nodes with declarations types equal to type.
+ * @param {[{node: ASTNode, declarationType: string}]} nodes An array contains objects, each object contains a node and a declaration type.
+ * @param {string} type Declaration type.
+ * @returns {[ASTNode]} An array contains only nodes with declarations types equal to type, if the nodes are null we return [].
+ */
+function getNodesByDeclarationTypeAndFormat(nodes, type) {
```

Comment:
What does "AndFormat" mean?

Reply:
The meaning is formatting the array after the filtering, instead of returning an array of objects, we return an array of nodes. But I removed it to make the name of the function simple.